smart-contract-extensions/contracts/hooks/Erc721ValidKeyHook.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.2;
+
+import '@unlock-protocol/contracts/dist/PublicLock/IPublicLockV9.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721.sol';
+
+contract Erc721TokenUriHook {
+  
+  mapping(address => address) nftAddresses;
+
+  function createMapping(
+    address _lockAddress, 
+    address _nftAddress
+  ) 
+  external 
+  {
+    // make sure lock manager
+    IPublicLockV9 lock = IPublicLockV9(_lockAddress);
```

Comment:
At some point we will need to find a way to have a "Generic" ABI that would work for all versions... because I assume this means it would break for v10 but I am pretty sure we will want to keep v10 compatible here?